oauth/client_test.go

```diff
@@ -0,0 +1,267 @@
+	stripe "github.com/stripe/stripe-go"
+	_ "github.com/stripe/stripe-go/testing"
+)
+
+func TestAuthorizeURL(t *testing.T) {
+	url := AuthorizeURL(&stripe.AuthorizeURLParams{})
+
+	assert.Equal(t, url, "https://connect.stripe.com/oauth/authorize?")
+}
+
+func TestAuthorizeURLWithOptionalArgs(t *testing.T) {
+	url := AuthorizeURL(&stripe.AuthorizeURLParams{
+		ClientID:              stripe.String("ca_123"),
+		State:                 stripe.String("test-state"),
+		Scope:                 stripe.String("read_only"),
+		RedirectURI:           stripe.String("https://t.example.com"),
+		ResponseType:          stripe.String("test-code"),
+		StripeLanding:         stripe.String("register"),
+		AlwaysPrompt:          stripe.Bool(true),
+		Express:               stripe.Bool(true),
+		SuggestedCapabilities: stripe.StringSlice([]string{"card_payments"}),
+	})
+
+	assert.Contains(t, url, "https://connect.stripe.com/express/oauth/authorize?")
+	assert.Contains(t, url, "client_id=ca_123")
+	assert.Contains(t, url, "redirect_uri=https%3A%2F%2Ft.example.com")
+	assert.Contains(t, url, "response_type=test-code")
+	assert.Contains(t, url, "scope=read_only")
+	assert.Contains(t, url, "state=test-state")
+	assert.Contains(t, url, "stripe_landing=register")
+	assert.Contains(t, url, "always_prompt=true")
+	assert.Contains(t, url, "suggested_capabilities[0]=card_payments")
+}
+
+func TestAuthorizeURLWithStripeUser(t *testing.T) {
+	url := AuthorizeURL(&stripe.AuthorizeURLParams{
+		ResponseType: stripe.String("test-code"),
+		StripeUser: &stripe.OAuthStripeUserParams{
+			BlockKana:          stripe.String("block-kana"),
+			BlockKanji:         stripe.String("block-kanji"),
+			BuildingKana:       stripe.String("building-kana"),
+			BuildingKanji:      stripe.String("building-kanji"),
+			BusinessName:       stripe.String("b-name"),
+			BusinessType:       stripe.String("llc"),
+			City:               stripe.String("Elko"),
+			Country:            stripe.String("US"),
+			Currency:           stripe.String("USD"),
+			DOBDay:             stripe.Int64(15),
+			DOBMonth:           stripe.Int64(10),
+			DOBYear:            stripe.Int64(2019),
+			Email:              stripe.String("test@example.com"),
+			FirstName:          stripe.String("first-name"),
+			FirstNameKana:      stripe.String("first-name-kana"),
+			FirstNameKanji:     stripe.String("first-name-kanji"),
+			Gender:             stripe.String("female"),
+			LastName:           stripe.String("last-name"),
+			LastNameKana:       stripe.String("last-name-kana"),
+			LastNameKanji:      stripe.String("last-name-kanji"),
+			PhoneNumber:        stripe.String("999-999-9999"),
+			PhysicalProduct:    stripe.Bool(false),
+			ProductDescription: stripe.String("product-description"),
+			State:              stripe.String("NV"),
+			StreetAddress:      stripe.String("123 main"),
+			URL:                stripe.String("http://example.com"),
+			Zip:                stripe.String("12345"),
```

Comment:
I just realized that "Zip" is actually an acronym (Zone Improvement Plan) and probably should've been "ZIP" instead. Don't change it though because we've already got the "Zip" convention in some fields elsewhere.

Reply:
Got it. Yeah tried to follow convention :)